relayer/src/services/rpc/merkleTreeService.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  IDL_LIGHT_MERKLE_TREE_PROGRAM,
+  Provider,
+  SolMerkleTree,
+  getRootIndex,
+  merkleTreeProgramId,
+} from "@lightprotocol/zk.js";
+import { getLightProvider } from "../../utils/provider";
+import { getIndexedTransactionsJob } from "./eventService";
+import { Program } from "@coral-xyz/anchor";
+import { PublicKey } from "@solana/web3.js";
+
+export const getMerkleRoot = async (req: any, res: any) => {
+  try {
+    console.log("@getMerkleRoot! ", req.body);
+    const { merkleTreePdaPublicKey: merkleTreePdaPublicKeyString } = req.body;
+
+    const merkleTreePdaPublicKey = new PublicKey(merkleTreePdaPublicKeyString);
+    const indexedTransactions = await getIndexedTransactionsJob(
+      merkleTreePdaPublicKey,
+    );
+    if (!indexedTransactions) {
+      // TODO: use different status code
+      return res
+        .status(200)
+        .json({ status: "error", message: "No indexed transactions found" });
+    }
+
+    const provider: Provider = await getLightProvider();
+    const merkleTreeProgram = new Program(
+      IDL_LIGHT_MERKLE_TREE_PROGRAM,
+      merkleTreeProgramId,
+      provider.provider,
+    );
+    const merkleTree = await SolMerkleTree.build({
+      pubkey: merkleTreePdaPublicKey,
+      hasher: provider.hasher,
+      indexedTransactions: indexedTransactions.map((trx) => trx.transaction),
+      provider: provider.provider,
+    });
+    // TODO: avoid rpc calls to accounts, stream the data of accounts and cache it instead
+    const index = await getRootIndex(
+      merkleTreeProgram,
+      merkleTree.pubkey,
+      merkleTree.merkleTree.root(),
+    );
+
+    return res.status(200).json({
+      data: { root: merkleTree.merkleTree.root(), index: index.toNumber() },
+    });
+  } catch (e) {
+    console.log(e);
+    return res.status(500).json({ status: "error", message: e.message });
+  }
+};
+
+export const getMerkleProofByIndexBatch = async (req: any, res: any) => {
+  try {
+    console.log("@getMerkleProofByIndexBatch! ", req.body);
+
+    const { merkleTreePdaPublicKey: merkleTreePdaPublicKeyString, indexes } =
+      req.body;
+    const merkleTreePdaPublicKey = new PublicKey(merkleTreePdaPublicKeyString);
+    let indexedTransactions = await getIndexedTransactionsJob(
+      merkleTreePdaPublicKey,
+    );
+    if (!indexedTransactions) {
+      // TODO: use different status code
+      // return res
+      //   .status(200)
+      //   .json({ status: "error", message: "No indexed transactions found"});
```

Comment:
should we uncomment this?

Reply:
No it would break the current implementation. I will remove this.